dialogue-core/src/main/java/com/palantir/dialogue/core/TraceEnrichingChannel.java

```diff
@@ -40,11 +41,18 @@
 
     @Override
     public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
+        if (!Tracer.isTraceObservable()) {
```

Comment:
I think we want to ensure observability has been decided.

Tracer.hasTraceId() && !Tracer.isTraceObservable()